p2p/src/peers/mod.rs

```diff
@@ -159,15 +277,18 @@ impl Peers {
 
     /// Add multiple peer addresses and save timestamp in the tried addresses bucket
     /// If an address did already exist, it gets overwritten
+    /// If an address is in the ice bucket, it gets ignored
     /// Returns all the overwritten or rejected addresses
     pub fn add_to_tried(
         &mut self,
         address: SocketAddr,
     ) -> Result<Option<SocketAddr>, failure::Error> {
-        // Insert address
-        let result = if !address.ip().is_unspecified() {
+        // Insert address, silently ignoring unspecified addresses and "iced" addresses
+        let result = if !address.ip().is_unspecified() && !self.ice_bucket_contains(&address) {
```

Comment:
Wondering whether this is something the feeler can do. Assuming peers in the iced bucket are not announced, if all peers have a particular peer in the ice bucket it might end up in the situation where the peer can not get out from the ice bucket

Reply:
Peers should be able get out of the ice bucket as soon as they get announced to someone that didn't iced them, or someone that iced them more than some certain time ago ("ice has melted").